node/src/types/deploy.rs

```diff
@@ -632,6 +650,32 @@ impl Deploy {
                 got: header.ttl(),
             });
         }
+        
+        let payment_args_length = self.payment().args().serialized_length();
+        if payment_args_length > config.payment_args_max_length as usize {
+            warn!(
```

Comment:
You should not log a warning for this (or below), it is a user error. Info or debug would be acceptable.